### PR TITLE
RAC-5938 :Refactor vagrant build shell script

### PR DESCRIPTION
**Pull request by PengTian0**
Implement a shell script to build vagrant box.

Test:
http://10.**.**.17*:8080/job/pengtian/job/BuildVagrantBox/

@panpan0000 @changev @anhou @iceiilin

### CODE REVIEW

File: packer/build_box.sh [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+#########################################
+#
+#  Usage
+#
+#########################################
+Usage(){
+    echo "Function: this script is used to build vagrant box"
+    echo "usage: $0 [arguments]"
+    echo "    mandatory arguments:"
+    echo "      --OS_VER: The version of "
```
Comment (panpan0000) [inline packer/build_box.sh]: uncompleted comment

File: packer/build_box.sh [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    set -e
+}
+
+build(){
+    pushd $DIR
+    echo "[Info] Generate a parameter file pass to packer"
+    CFG_FILE=template.cfg
+    # parameter file pass to packer
+    echo {                                                        > $CFG_FILE
+    echo  \"playbook\": \"rackhd_ci_builds_mini\",                >> $CFG_FILE
```
Comment (panpan0000) [inline packer/build_box.sh]: how abort ```rackhd_release_mini.yml```?

File: packer/build_box.sh [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    # pre-process the packer template file
+    PACKER_TEMP=template-${OS_VER}.json.tmp
+    TMP_FILE_STREAM=$(cat template-${OS_VER}.json)
+    # Delete the post-processors blocks in the template.json, before sending to 'packer build'
+    TMP_FILE_STREAM=$( jq 'del(.["post-processors"][0][1])' template-${OS_VER}.json | jq 'del(.["push"])')
+    # Write back the template file
+    echo "$TMP_FILE_STREAM" > $PACKER_TEMP
+    echo "[Info] The template-${OS_VER}.json has been customized"
+
+    if [ -n "$CACHE_IMAGE_DIR" ];then
```
Comment (panpan0000) [inline packer/build_box.sh]: if [ *!* -n "$CACHE_IMAGE_DIR" ];then ???

File: packer/build_box.sh (hunk 1 above)
Comment (PengTian0) [reply packer/build_box.sh]: Thanks!
Fixed.

File: packer/build_box.sh (hunk 2 above)
Comment (PengTian0) [reply packer/build_box.sh]: The ```rackhd_ci_builds_mini.yml``` and ```rackhd_release_mini.yml``` both work.
Do you mean using ```rackhd_release_mini.yml``` here?

**Pull request merged**